### DISCUSSION TITLE
Add video and music downloads

### DISCUSSION THREAD

**Discussion by XRedCubeX** [ReVanced/revanced-discussions#117] [Question] [answered]
This is a more than legitimate question, given the fact that Vanced has tried to avoid problems by not inserting this function but despite not having included it, but in any case they have had a warning/complaint.

**Comment by oSumAtrIX** [comment, marked as answer]
I recommend checking out the #faq channel in our Discord server. A patch can be created to enable downloads. The entire system is designed in a way, that you could fork the patches repository, create your own patch and use it yourself. So either someone creates an "unofficial" patch which peope can apply, or we include it in our repository

**Comment by XRedCubeX** [comment]
> I recommend checking out the #faq channel in our Discord server. A patch can be created to enable downloads. The entire system is designed in a way, that you could fork the patches repository, create your own patch and use it yourself. So either someone creates an "unofficial" patch which peope can apply, or we include it in our repository

I looked at the FAQ and understood the system, thanks for the answer, have a nice day!

**Comment by salman0ansari** [comment]
why do we need video downloader?

**Comment by oSumAtrIX** [comment]
> why do we need video downloader?

To download and watch videos offline when outside or on a trip etc

**Comment by MSOB7YY** [comment]
> > why do we need video downloader?
> 
> To download and watch videos offline when outside or on a trip etc

i would prefer something like permanent caching, saving the video and webm page (description, comments, channel, etc) and then playing within the app itself

> also getting other cached video as  recommendations would be awesome if possible-

**Comment by oSumAtrIX** [comment]
Well yeah, thats what the download button does on the YouTube app, the patch would simply enable it

**Comment by MSOB7YY** [comment]
> Well yeah, thats what the download button does on the YouTube app, the patch would simply enable it

damn that would be just ooooff🔥🔥

**Comment by XRedCubeX** [comment]
> Well yeah, thats what the download button does on the YouTube app, the patch would simply enable it

This is exactly what I meant

**Comment by Sculas** [comment]
Don't close this issue, this is the patch request for video downloads.

**Comment by XRedCubeX** [comment]
I'm sorry, by mistake instead of clicking on comment I clicked on close and comment

**Comment by DraakjeYoblama** [comment]
> I recommend checking out the #faq channel in our Discord server. A patch can be created to enable downloads. The entire system is designed in a way, that you could fork the patches repository, create your own patch and use it yourself. So either someone creates an "unofficial" patch which peope can apply, or we include it in our repository

I don't know how this works legally speaking, but I would be hesitant to include it in the official repository. I think those youtube downloader things get taken down quite often and you wouldn't want to give Google a reason to close down revanced.
Maybe there's no legal issue with it, then just ignore this.

**Comment by Degl1** [comment]
NewPipe and VLC Media Player work fine just fine.

**Comment by Utku-I** [comment]
Yeah, but the download functionality on the YouTube app is kinda different, since you can see comments. Also, would be nice since it is faster to click download rather than share and newpipe.

> NewPipe and VLC Media Player work fine just fine.

**Comment by XRedCubeX** [comment]
I've seen the latest commits and I'm glad this feature is under construction and I'm curious to know: will this download feature be ported to Youtube Music as well?
